### PR TITLE
Experimental: Test GitHub Actions locally

### PR DESCRIPTION
**Pull request by stefaniuk**
## Description

This PR introduces the `nektos/act` project to run GitHub Actions workflows on a developer's local machine, providing a fast, cost-effective and resource-efficient way to test changes to actions and debug before pushing to GitHub. It emulates the GitHub Actions runner environment using Docker containers, allowing for an identical testing conditions.

## Context

image

### Testing

```
# clone the repo and switch to the `Test_GitHub_Actions_locally` branch, then run:
$ make runner-act workflow="stage-1-commit" job="cloc-repository"
```

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by timrickwood — APPROVED** [review]

**Review by jacobgill1 — APPROVED** [review]
LGTM

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-yyyymmdd
+${yyyy}${mm}${dd}
```
Comment (amaanibn-nasar1-nhs) [inline VERSION:1]: Not sure entirely how this works but do we want to have slashes (/) or dashes (-) for clarity of reading?

**Review by amaanibn-nasar1-nhs — COMMENTED** [review]
One question

File: VERSION (hunk 1 above)
Comment (stefaniuk) [reply VERSION:1]: That's a good observation. During the build process, a date like `20230914` might be produced. This approach to versioning artefacts, known as calendar versioning, is widely recognised. To enhance clarity and align it more closely with semantic versioning, the format `${yyyy}.${mm}.${dd}` is suggested, which would translate to `2023.09.14`. Both formats are supported by our template. It's up to individual engineering teams to decide which format suits their repositories best.

For our repository template, we'll adopt either the `${yyyy}.${mm}.${dd}` or the `x.y.z` version format. But this is yet to be decided. Thus, this file will undergo change again before producing the release candidate.

As part of this PR we wanted to introduce more intuitive and less error prone pattern `${...}`

**Review by amaanibn-nasar1-nhs — APPROVED** [review]
LGTM

**Pull request merged**